### DISCUSSION TITLE
PackerCompile requires relaunching neovim?

### DISCUSSION THREAD

**Discussion by cultab** [wbthomason/packer.nvim#1118] [Q&A]
I'm just trying to see if I have this right in my head cause I lost about 2 hours of my time on this haha.

You have a plugin and want to give it a `config` function to `setup()` a plugin with some options.
You give it said function by requiring a module that returns a config function (or a table with a `config` key or whatever),
Then running `:PackerCompile` uses whatever was returned from requiring the module the **first time**, since requires are cached EVEN if you then change the code in said module.

Meaning that if you then change the config function which resides in another module, it won't actually be used until neovim restarts or you unload the module?

And this would be solved by running the returned config function in a closure or having the module just run the configuration code itself?

**Comment by nyngwang** [comment]
I don't fully understand your description, but I do know that I always have to restart neovim after calling `:PackerSync`. Without restatring, renewal of the file `plugins/packer_compiled.lua` doesn't work.
